src/Proxy.NET/Authentication/JwtAuthenticationHandler.cs

```diff
@@ -0,0 +1,35 @@
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Options;
+using Proxy.NET.Services;
+
+namespace Proxy.NET.Authentication;
+
+public class JwtAuthenticationHandler(
+    IOptionsMonitor<ProxyAuthenticationOptions> options,
+    IAuthorizeService authorizeService,
+    ILoggerFactory logger,
+    UrlEncoder encoder
+) : AuthenticationHandler<ProxyAuthenticationOptions>(options, logger, encoder)
+{
+    protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        if (!Request.Headers.TryGetValue("x-ms-client-principal", out var jwt) || string.IsNullOrEmpty(jwt))
+            return AuthenticateResult.Fail("Authentication failed.");
+
+        var requestContext = authorizeService.GetRequestContextFromJwt(jwt!);
```

Comment:
```suggestion
        var requestContext = authorizeService.GetRequestContextFromJwt(jwt);
```

The compiler should detect that `jwt` is not `null` due to the check above.